bundles/org.openhab.binding.automower/src/main/java/org/openhab/binding/automower/internal/things/AutomowerHandler.java

```diff
@@ -217,27 +215,33 @@ public void initialize() {
                 mowerZoneId = timeZoneProvider.getTimeZone(); // not configured, use System TimeZone
             }
 
-            if (mowerId == null) {
-                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
-                        "@text/conf-error-no-mower-id");
-            } else {
-                automowerId.set(mowerId);
-                // Adding handler to map of handlers
-                AutomowerBridgeHandler automowerBridgeHandler = getAutomowerBridgeHandler();
-                if (automowerBridgeHandler != null) {
-                    automowerBridgeHandler.registerAutomowerHandler(mowerId, this);
-                }
-
-                // initial poll to get the current state of the mower
-                poll();
-                // update messages once via polling of REST API and later event based via WebSocket only
-                initializeMessages(mowerId);
+            final String mowerId = this.getThing().getUID().getId();
+            automowerId.set(mowerId);
+            // Adding handler to map of handlers
+            AutomowerBridgeHandler automowerBridgeHandler = getAutomowerBridgeHandler();
+            if (automowerBridgeHandler != null) {
+                automowerBridgeHandler.registerAutomowerHandler(mowerId, this);
             }
+
+            updateStatus(ThingStatus.UNKNOWN); // Set to UNKNOWN initially
+
+            scheduler.execute(() -> completeInitAsync());
+
         } else {
             updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_UNINITIALIZED);
         }
     }
 
+    private void completeInitAsync() {
+        String mowerId = automowerId.get();
+        if (!mowerId.equals(NO_ID)) {
+            // initial poll to get the current state of the mower
+            poll();
```

Comment:
I'm missing a path that sets the status to online. See other handler for maybe returning boolean from poll and set online state based on that boolean?

Reply:
poll triggers `updateAutomowerState()` which sets it `ONLINE`:

```
    private void updateAutomowerState(Mower mower) {
        this.mowerState = mower;
        if (isValidResult(this.mowerState)) {
            initializeProperties(mower);
            updateMowerChannelState(mower);
            if (isConnected(mower)) {
                updateStatus(ThingStatus.ONLINE);
            }
```